### PR TITLE
EVM: basic implementation of `eth_sendRawTransaction `

### PR DESCRIPTION
**Pull request by bkolad**
# Description
This PR introduces the following changes:
1. `sov-etehreum` crate in `full-node`. This crate provides `eth_sendRawTransaction ` rpc endpoint. This endpoint wraps evm transaction in rollup transaction and sends it to celestia (the endpoint will be generic over DA in the future).
Currently, the implementation is very basic. The logic panics if unexpected data is encountered, and some of the values in the request are hardcoded to defaults. Despite these limitations, it is sufficient for a simple unit test to function and will be improved in future pull requests.

2. Adds a couple of ethereum rpc endpoints in `sov-evm/src/query.rs`. Once again, the implementation is minimal and will be expanded upon in the next pull requests.

## Linked Issues
- Related to #413

## Testing
Tested the new endpoint manually.

### CODE REVIEW

File: examples/demo-rollup/rollup_config.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ start_height = 1
 
 [da]
 # The JWT used to authenticate with the celestia light client. Instructions for generating this token can be found in the README
-celestia_rpc_auth_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJwdWJsaWMiLCJyZWFkIiwid3JpdGUiLCJhZG1pbiJdfQ.YNNrJz6mCD6ypuH9Ay6NuCy1B6oQVTa2wi5-etSeY-8"
+celestia_rpc_auth_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJwdWJsaWMiLCJyZWFkIiwid3JpdGUiLCJhZG1pbiJdfQ.yEqKNCDpXwddRTQYQeMNkQ59O22LI95CS-HeZMcoN-k"
```
Comment (citizen-stig) [inline examples/demo-rollup/rollup_config.toml:7]: I guess we keep having this problem.

File: full-node/sov-ethereum/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        );
+
+        jsonrpsee::http_client::HttpClientBuilder::default()
+            .set_headers(headers)
+            .max_request_body_size(default_max_response_size()) // 100 MB
+            .build(self.config.celestia_rpc_address.clone())
+            .expect("Client initialization is valid")
+    }
+
+    async fn send_tx_to_da(
```
Comment (citizen-stig) [inline full-node/sov-ethereum/src/lib.rs:46]: Just an idea, but since Sequencer is merged, it can be used to create blobs.

Comment (citizen-stig) [reply full-node/sov-ethereum/src/lib.rs:46]: Also, this functionality is available in DaLaayer service. So current implementation depends on Celestia, where if it uses trait, it should work on all Da Layers.

File: full-node/sov-ethereum/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+fn register_rpc_methods(rpc: &mut RpcModule<Ethereum>) -> Result<(), jsonrpsee::core::Error> {
+    rpc.register_async_method(
+        "eth_sendRawTransaction",
+        |parameters, ethereum| async move {
+            let data: Bytes = parameters.one().unwrap();
+            let data = data.as_ref();
+
+            // todo handle panics and unwraps.
+            if data[0] > 0x7f {
+                panic!("Invalid transaction type")
```
Comment (citizen-stig) [inline full-node/sov-ethereum/src/lib.rs:76]: Woud `anyhow::bail` work here?

File: full-node/sov-ethereum/src/lib.rs (hunk 2 above)
Comment (bkolad) [reply full-node/sov-ethereum/src/lib.rs:46]: Yes I will extract the common functionality in the next PR.

File: full-node/sov-ethereum/src/lib.rs (hunk 3 above)
Comment (bkolad) [reply full-node/sov-ethereum/src/lib.rs:76]: Here we will need to craft a proper `jsonrpsee` response, the same way Ethereum does it. But it is out of scope for now.

File: examples/demo-rollup/rollup_config.toml (hunk 1 above)
Comment (bkolad) [reply examples/demo-rollup/rollup_config.toml]: Removed the file and added to .gitignore.

**Pull request closed without merging**